frappe/core/doctype/doctype/doctype.py

```diff
@@ -1443,6 +1431,23 @@ def check_no_of_ratings(docfield):
 			if docfield.options and (int(docfield.options) > 10 or int(docfield.options) < 3):
 				frappe.throw(_("Options for Rating field can range from 3 to 10"))
 
+	def validate_default_values(docfield):
+		if docfield.get("default"):
+			if docfield.fieldtype == "Int":
+				if not isinstance(docfield.default, int):
+					frappe.throw(
+						_("Default value for field {0} must be an integer").format(frappe.bold(docfield.label)),
+						title=_("Invalid Value"),
+					)
+			elif docfield.fieldtype in ["Percent", "Float"]:
+				try:
+					float(docfield.default)
+				except ValueError:
+					frappe.throw(
+						_("Default value for field {0} must be a number").format(frappe.bold(docfield.label)),
+						title=_("Invalid Value"),
+					)
```

Comment:
this is too noisy - can we make the error generic where we just need to pass in the type expected? otherwise we'll be creating unnecessary `try except` blocks for every `if` branch

Also, there is a function `check_illegal_default` - which i think is for this validation itself - maybe we can club these both?

Reply:
Have moved this to `check_illegal_default`, not sure what you meant by generic error. Int and Float fields have to be validated separately, instead of here the generic function will have the try-catch or branching